### ISSUE TITLE
H00052: child pid x exit signal Abort trap (6)

### ISSUE THREAD

**Issue by glassdimly** [Homebrew/homebrew-php#4773]
## Description

On certain PHP pages, I get this in my apache logs:

```
[Fri Feb 16 11:36:11.748594 2018] [core:notice] [pid 19652] AH00052: child pid 21236 exit signal Abort trap (6)
[Fri Feb 16 11:36:11.748610 2018] [core:notice] [pid 19652] AH00052: child pid 21235 exit signal Abort trap (6)
[Fri Feb 16 11:36:11.748625 2018] [core:notice] [pid 19652] AH00052: child pid 21234 exit signal Abort trap (6)
[Fri Feb 16 11:36:11.748641 2018] [core:notice] [pid 19652] AH00052: child pid 21233 exit signal Abort trap (6)
Assertion failed: (0), function check_type_narrowing, file /private/tmp/php71-opcache-20170209-90894-15sn9lw/php-7.1.1/ext/opcache/Optimizer/zend_inference.c, line 1894.
[Fri Feb 16 11:36:12.750371 2018] [core:notice] [pid 19652] AH00052: child pid 21634 exit signal Abort trap (6)
Assertion failed: (0), function check_type_narrowing, file /private/tmp/php71-opcache-20170209-90894-15sn9lw/php-7.1.1/ext/opcache/Optimizer/zend_inference.c, line 1894.
Assertion failed: (0), function check_type_narrowing, file /private/tmp/php71-opcache-20170209-90894-15sn9lw/php-7.1.1/ext/opcache/Optimizer/zend_inference.c, line 1894.
[Fri Feb 16 11:36:13.754022 2018] [core:notice] [pid 19652] AH00052: child pid 21637 exit signal Abort trap (6)
[Fri Feb 16 11:36:13.754484 2018] [core:notice] [pid 19652] AH00052: child pid 21636 exit signal Abort trap (6)
```

I really don't know where to start with this issue. I tried disabling my custom `php.ini` config, but that didn't help. I realize this might not be the right forum for this issue, but if you could point me in the right direction, or even give me a little context about what this error means, I'd be grateful.

I specifically get this error on this Drupal 8 page, across all sites: `mysite.dev/admin/structure/types/manage/{my_content_type}/display`

## Context and Logs

Parameter | Value
--------- | ------------------
**macOS version:**       _10.13.2_
**Homebrew version:**   _Homebrew/homebrew-core (git revision f938c; last commit 2018-02-07)_
**PHP version in use:** _7.1.1_
**Xcode version:**   _9.2_

**Output of `gcc -v`:**
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk/usr/include/c++/4.2.1
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin17.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

**Output of `php -v`:**
PHP 7.1.1 (cli) (built: Feb  9 2017 11:11:52) ( ZTS DEBUG )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.1, Copyright (c) 1999-2017, by Zend Technologies
    with Xdebug v2.5.0, Copyright (c) 2002-2016, by Derick Rethans

**Build logs:** https://gist.github.com/anonymous/1a692f30d04b56832e4ae4ecaaf90416

**Comment by glassdimly** [comment]
Got any tips for a friend, @alanthing?

**Comment by glassdimly** [comment]
Thanks for your response. OK, I'm upgrading all the things. I hesitate because it seems like often when I upgrade stuff breaks.

**Comment by glassdimly** [comment]
(I did upgrade everything, and everything did break. That's because I'm using `--with-thread-safety` for redis and php and all modules have to re-compile.

Then, also, I'm using symlinks to an etc directory so I can version stuff, and some of the brew build scripts don't follow symlinks.

Anyways.)

That did fix it. Things I did, in case someone encounters this thread:

* upgraded php: `7.1` -> `7.1.14`
* Moved from the `httpd24` module to `httpd`
* Using MPM worker `mod_mpm_prefork.so` because it's just localhost. I don't know which MPM I was previously using.
